# Tâche n°5 : Détermination de différentes caractéristiques du matériau.

Tout d'abord, les informations sur la structure du matériau sont téléchargées comme fait à la tâche précédente.

In [26]:
from pymatgen.ext.matproj import MPRester
from pymatgen.ext.matproj import SpacegroupAnalyzer

with MPRester("xWGoUhg0LtBEwfAi") as m:
    structure = m.get_structure_by_material_id("mp-754537")
    truestru = SpacegroupAnalyzer(structure).get_primitive_standard_structure() 
    sg = SpacegroupAnalyzer(truestru)
    print(truestru)

Full Formula (Li1 Lu1 O2)
Reduced Formula: LiLuO2
abc   :   5.375778   5.375778   5.375778
angles:  35.456019  35.456019  35.456019
Sites (4)
  #  SP           a         b         c
---  ----  --------  --------  --------
  0  Li    0         0         0
  1  Lu    0.5       0.5       0.5
  2  O     0.243887  0.243887  0.243887
  3  O     0.756113  0.756113  0.756113


## 1) Vecteur de bases réseaux direct 

In [15]:
rdm = truestru.lattice # vecteurs de base du réseaux direct sous forme matriciel
rd = rdm.abc #vecteur de base du réseaux direct sous forme abc

print("Vecteurs de base du réseau direct sous forme matriciel :", rdm)
print("Vecteurs de base du réseau direct sous forme abc :", rd)


Vecteur de base du réseau direct sous forme matriciel : 5.120497 -1.636918 0.000000
5.120497 1.636918 0.000000
4.597208 0.000000 2.786515
Vecteur de base du réseau direct sous forme abc : (5.375778177650529, 5.375778177650529, 5.375778177650529)


On peut vérifier que les vecteurs de base du réseaux direct sont correctes en les comparant aux informations données par la variable "truestru" affichées ci-dessus.

## 2) Vecteur de bases réseaux réciproques

In [25]:
rrm = truestru.lattice.reciprocal_lattice # vecteurs de base du réseau réciproque sous forme matriciel
rr = rrm.abc # vecteurs de base du réseau réciproquesous forme abc

print("Vecteurs de base du réseaux réciproque sous forme matriciel :",rrm)
print("Vecteurs de base du réseaux réciproque sous forme abc :",rr)


Vecteur de base du réseaux réciproque sous forme matriciel : 0.613533 -1.919212 -1.012210
0.613533 1.919212 -1.012210
0.000000 0.000000 2.254854
Vecteur de base du réseaux réciproque sous forme abc : (2.25485411757049, 2.25485411757049, 2.2548541175704897)


La vérification de ces vecteurs de bases peut être éffectuée en prenant la définition que : *a*⋅a= 2π*. Si on fait le produit scalaire entre a = [5.120497 -1.636918 0.000000] et a* = [0.613533 -1.919212 -1.012210], celui-ci est bien égale à *2π*.


## 3) Type de maille

In [17]:
tmaille = sg.get_lattice_type()
print("Son type de maille est" ,tmaille)

Son type de maille est rhombohedral


## 4) Système cristallin

In [18]:
sc = sg.get_crystal_system()
print("Son système cristallin est", sc) 


Son système cristallin est trigonal


## 5) Groupe ponctuel
Avec la notation de Hermann Mauguin

In [19]:
gp = sg.get_point_group_symbol() 
print("Son groupe ponctuel est", gp)

Son groupe ponctuel est -3m


# Références

- [Materials Project](https://materialsproject.org) 
- [pymatgen.core.structure](https://pymatgen.org/pymatgen.core.structure.html) Ce lien redirige vers l'interface "pymatgen.core.structure" permettant de trouver l'existence de la propriétée structure.lattice (Utile pour le point 1 et 2)
- [pymatgen.core.lattice](https://pymatgen.org/pymatgen.core.lattice.html) Ce lien redirige vers l'interface "pymatgen.core.lattice" permettant de trouver l'existence des propriétés : "Lattice.abc" et "Lattice.reciprocal_lattice" (Utile pour les points 1 et 2)
- [pymatgen.symmetry.analyzer](https://pymatgen.org/pymatgen.symmetry.analyzer.html?highlight=spacegroupanalyzer#pymatgen.symmetry.analyzer.SpacegroupAnalyzer) Ce lien redirige vers l'interface "pymatgen.symmetry.analyzer" permettant de trouver les spécifications des trois fonctions utilisées pour répondre aux points 3-4-5.